### PR TITLE
[BugFix] `GroupReaderParam.Column.content_type` is modified by multiple group readers

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

This bug happens when:
1. A file reader has multiple group readers, and
2. In one row group, a column is encoded as dictionary page
3. But in another row group, a column is encoded as plain page.

We suppose `_param.read_cols` should be immutable(and it's shared ny multiple group readers). However it's modified by each group reader. 

```
    for (auto& column : _param.read_cols) {
        int chunk_index = column.col_idx_in_chunk;
        SlotId slot_id = column.slot_id;
        const tparquet::ColumnMetaData& column_metadata =
                _row_group_metadata->columns[column.col_idx_in_parquet].meta_data;
        if (_can_using_dict_filter(slots[chunk_index], conjunct_ctxs_by_slot, column_metadata)) {
            column.content_type = ColumnContentType::DICT_CODE;
           ...
        } else {
            column.content_type = ColumnContentType::VALUE;
            ...
```

TODO: try to reproduce this case and add UT.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: be/src/formats/parquet/group_reader.h [hunk 1]
```diff
@@ -121,11 +118,11 @@ class GroupReader {
     // conjunct ctxs that eval after chunk is dict decoded
     std::vector<ExprContext*> _left_conjunct_ctxs;
 
-    // dict filter column
-    std::vector<GroupReaderParam::Column> _dict_filter_columns;
-    // direct read conlumns
-    std::vector<GroupReaderParam::Column> _direct_read_columns;
-
+    // dict filter column index
```
Comment (DorianZheng) [inline be/src/formats/parquet/group_reader.h:121]: Please create a class to encapsulate dict filter column information. because we don't want it to be visible everywhere

File: be/src/formats/parquet/group_reader.h [hunk 2]
```diff
@@ -121,11 +118,11 @@ class GroupReader {
...
+    std::vector<int> _dict_filter_column_indices;
+    std::vector<bool> _use_as_dict_filter_column;
+    // direct read conlumn index
+    std::vector<int> _direct_read_column_indices;
```
Comment (GavinMar) [inline be/src/formats/parquet/group_reader.h:125]: What is the purpose of this change? To reduce memory and copy overhead? It seems it is not related with the bug?

**Review by DorianZheng — DISMISSED** [review]

**Review by GavinMar — DISMISSED** [review]

**Review by satanson — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by dirtysalt** [comment]
@MergifyIO backport branch-2.5
